src/overlays/actors/ovl_En_Elforg/z_en_elforg.h

```diff
@@ -2,36 +2,41 @@
 #include "objects/gameplay_keep/gameplay_keep.h"
 
 #define STRAY_FAIRY_TYPE(thisx) ((thisx)->params & 0xF)
 #define STRAY_FAIRY_GET_NON_DUNGEON_AREA(thisx) (((thisx)->params & 0x1C0) >> 6)
 #define STRAY_FAIRY_FLAG(thisx) (((thisx)->params & 0xFE00) >> 9)
-#define STRAY_FAIRY_PARAMS(flag, nonDungeonArea, type) (((flag & 0x7F) << 9) | ((nonDungeonArea & 7) << 6) | (type & 0xF))
+#define STRAY_FAIRY_SPARKLE_COUNT(thisx) ((thisx)->home.rot.x)
+
+//! @note `nonDungeonArea` does not always use the enum
+#define STRAY_FAIRY_PARAMS(flag, nonDungeonArea, type) ((((flag) & 0x7F) << 9) | (((nonDungeonArea) & 7) << 6) | ((type) & 0xF))
 
 #define STRAY_FAIRY_FLAG_MOVES_QUICKLY_TO_HOME (1 << 0)
 #define STRAY_FAIRY_FLAG_SPARKLES_AND_SHRINKS (1 << 1)
 #define STRAY_FAIRY_FLAG_CIRCLES_QUICKLY_IN_FOUNTAIN (1 << 2)
 #define STRAY_FAIRY_FLAG_GREAT_FAIRYS_MASK_EQUIPPED (1 << 3)
 
-typedef enum {
-    /* 0 */ STRAY_FAIRY_TYPE_FREE_FLOATING,            // The ones just floating around
-    /* 1 */ STRAY_FAIRY_TYPE_FAIRY_FOUNTAIN,           // The ones already present when you enter a Fairy Fountain
-    /* 2 */ STRAY_FAIRY_TYPE_BUBBLE,                   // The ones trapped in bubbles
-    /* 3 */ STRAY_FAIRY_TYPE_CLOCK_TOWN,               // The free-floating Stray Fairies in Clock Town
-    /* 4 */ STRAY_FAIRY_TYPE_ENEMY,                    // The ones trapped inside enemies
-    /* 5 */ STRAY_FAIRY_TYPE_COLLIDER,                 // Unused in retail. The fairy is hidden until the collider is hit
-    /* 6 */ STRAY_FAIRY_TYPE_CHEST,                    // The ones in treasure chests
-    /* 7 */ STRAY_FAIRY_TYPE_COLLECTIBLE,              // The ones in boxes, pots, beehives, etc.
-    /* 8 */ STRAY_FAIRY_TYPE_TURN_IN_TO_FAIRY_FOUNTAIN // The ones you "turn in" by walking into a Fairy Fountain
+typedef enum StrayFairyType {
+    /* 0 */ STRAY_FAIRY_TYPE_FREE_FLOATING,        // The ones just floating around
+    /* 1 */ STRAY_FAIRY_TYPE_FAIRY_FOUNTAIN,       // The ones already present when you enter a Fairy Fountain
+    /* 2 */ STRAY_FAIRY_TYPE_BUBBLE,               // The ones trapped in bubbles
+    /* 3 */ STRAY_FAIRY_TYPE_CLOCK_TOWN,           // The free-floating Stray Fairies in Clock Town
+    /* 4 */ STRAY_FAIRY_TYPE_ENEMY,                // The ones trapped inside enemies
+    /* 5 */ STRAY_FAIRY_TYPE_COLLIDER,             // Unused in retail. The fairy is hidden until the collider is hit
+    /* 6 */ STRAY_FAIRY_TYPE_CHEST,                // The ones in treasure chests
+    /* 7 */ STRAY_FAIRY_TYPE_COLLECTIBLE,          // The ones in boxes, pots, beehives, etc.
+    /* 8 */ STRAY_FAIRY_TYPE_RETURNING_TO_FOUNTAIN // The ones you "turn in" by walking into a Fairy Fountain
 } StrayFairyType;
 
-typedef enum {
-    /* 0 */ STRAY_FAIRY_AREA_CLOCK_TOWN,
-    /* 1 */ STRAY_FAIRY_AREA_WOODFALL,
-    /* 2 */ STRAY_FAIRY_AREA_SNOWHEAD,
-    /* 3 */ STRAY_FAIRY_AREA_GREAT_BAY,
-    /* 4 */ STRAY_FAIRY_AREA_STONE_TOWER,
+// Correspond to the Great Fairy types
```

Comment:
Extremely pedantic nit, but:
```suggestion
// Corresponds to the Great Fairy types
```